pkg/registry/file/dynamicpathdetector/collapse_config_from_crd.go

```
@@ -0,0 +1,100 @@
+}
+
+// CRDFromCollapseSettings is the inverse of CollapseSettingsFromCRD. It
+// produces a fresh CollapseConfiguration suitable for client-go Create /
+// Update calls. Tooling (notably bobctl autotune) uses it to push tuned
+// thresholds back into a running cluster.
+func CRDFromCollapseSettings(name string, settings CollapseSettings) *softwarecomposition.CollapseConfiguration {
+	entries := make([]softwarecomposition.CollapseConfigEntry, len(settings.CollapseConfigs))
+	for i, cfg := range settings.CollapseConfigs {
+		entries[i] = softwarecomposition.CollapseConfigEntry{
+			Prefix:    cfg.Prefix,
+			Threshold: int32(cfg.Threshold),
```

golangci-lint (2.11.4)

[high] G115: integer overflow conversion int -> int32

(gosec)